src/ziggurat/messaging/consumer.clj

```diff
@@ -1,30 +1,35 @@
 (ns ziggurat.messaging.consumer
-  (:require [ziggurat.mapper :as mpr]
-            [clojure.tools.logging :as log]
+  (:require [clojure.tools.logging :as log]
             [langohr.basic :as lb]
             [langohr.channel :as lch]
             [langohr.consumers :as lcons]
-            [ziggurat.kafka-consumer.consumer-handler :as ch]
             [taoensso.nippy :as nippy]
-            [ziggurat.config :refer [get-in-config]]
+            [ziggurat.config :refer [get-in-config rabbitmq-config]]
+            [ziggurat.kafka-consumer.consumer-handler :as ch]
+            [ziggurat.mapper :as mpr]
             [ziggurat.messaging.connection :refer [connection]]
-            [ziggurat.sentry :refer [sentry-reporter]]
-            [ziggurat.messaging.util :refer :all]
+            [ziggurat.messaging.util :as util]
             [ziggurat.metrics :as metrics]
             [ziggurat.util.error :refer [report-error]]))
 
 (defn convert-and-ack-message
   "De-serializes the message payload (`payload`) using `nippy/thaw` and converts it to `MessagePayload`. Acks the message
   if `ack?` is true."
-  [ch {:keys [delivery-tag] :as meta} ^bytes payload ack? topic-entity]
+  [ch {:keys [delivery-tag]} ^bytes payload ack? topic-entity]
   (try
     (let [message (nippy/thaw payload)]
       (when ack?
         (lb/ack ch delivery-tag))
       message)
     (catch Exception e
-      (lb/reject ch delivery-tag false)
-      (report-error e "Error while decoding message")
+      (report-error e "Error while decoding message, publishing to dead queue...")
+      (let [{:keys [exchange-name]} (:dead-letter (rabbitmq-config))
```

Comment:
we can extract this entire code for "publish-to-deadset" into a method.  can  be reused for handling error in `process-message-from-queue` too

Reply:
done